### PR TITLE
chore: Update React to ^18.2.0 in  peerDependencies

### PR DESCRIPTION
**Pull request by patrykboch**
Allow React 18 to be used as a peer dependency. Fixes https://github.com/react-tags/react-tags/issues/855

### CODE REVIEW

**Comment by patrykboch** [comment]
@ad1992 please review

**Comment by ad1992** [comment]
@patrykboch can you also update package-lock.json file ?

**Comment by CalebJamesStevens** [comment]
@ad1992 @christowiz please stop ignoring this

**Comment by patrykboch** [comment]
I'll take care of this very soon.

**Review by spider-g — APPROVED** [review]

**Comment by patrykboch** [comment]
> @patrykboch can you also update package-lock.json file ?

@ad1992 Could you tell me what you mean, please? I don't see `package-lock.json` but `yarn.lock` instead. Should it be generated or maybe you mean to update react that is used internally by the lib?

**Comment by ad1992** [comment]
@patrykboch the `yarn.lock` file changes look too huge.
Can you do the below steps to update the file
```
Restore the original yarn.lock file from master branch
Run `yarn` in local
```
this should correctly update the `yarn` file and push it in your branch

**Comment by patrykboch** [comment]
I updated only peer dependencies, the dev dependencies still use react v. 17. Running yarn locally does not make a diff in `yarn.lock`, what we're missing?

File: package.json [hunk 1]
```diff
@@ -35,10 +35,10 @@
   "license": "MIT",
   "repository": "https://github.com/react-tags/react-tags",
   "peerDependencies": {
-    "react": "^17.0.2",
+    "react": "^17.0.2 || ^18.1.0",
```
Comment (ad1992) [inline package.json:38]:
```suggestion
    "react": "^17.0.2 || ^18.2.0",
```

File: package.json [hunk 2]
```diff
@@ -35,10 +35,10 @@
...
     "react-dnd": "^14.0.2",
     "react-dnd-html5-backend": "^14.0.0",
-    "react-dom": "17.0.2"
+    "react-dom": "17.0.2 || ^18.1.0"
```
Comment (ad1992) [inline package.json:41]:
```suggestion
    "react-dom": "17.0.2 || ^18.2.0"
```

**Comment by ad1992** [comment]
> I updated only peer dependencies, the dev dependencies still use react v. 17. Running yarn locally does not make a diff in `yarn.lock`, what we're missing?

Thats right, we use `react` and `react-dom` from cdn in our example so will need to update the cdn links in https://github.com/react-tags/react-tags/blob/master/example/index.html#L17

**Comment by apepper** [comment]
Awesome, thanks to @patrykboch and @ad1992 for taking care of this!

@ad1992 Any ETA when this change will be available on https://www.npmjs.com/package/react-tag-input?

**Comment by patrykboch** [comment]
@ad1992 I can't wait for this to be released :-) Thank you for merging this.

**Comment by apepper** [comment]
@mrdannael and @prakhar1989: Any chance you could do a new release of https://www.npmjs.com/package/react-tag-input? At least our team is patiently waiting for React 18 support (which is already merge in `master`).

**Comment by apepper** [comment]
@ad1992 Thank you so much!

**Pull request merged**